packages/material-ui/src/SliderUnstyled/SliderUnstyled.d.ts

```diff
@@ -0,0 +1,223 @@
+import { OverridableComponent } from '../OverridableComponent';
+
+export interface Mark {
+  value: number;
+  label?: React.ReactNode;
+}
+
+export interface ValueLabelProps extends React.HTMLAttributes<HTMLSpanElement> {
+  children: React.ReactElement;
+  index: number;
+  open: boolean;
+  value: number;
+}
+
+export interface SliderTypeMap<P = {}, D extends React.ElementType = 'span'> {
+  props: P & {
+    /**
+     * The label of the slider.
+     */
+    'aria-label'?: string;
+    /**
+     * The id of the element containing a label for the slider.
+     */
+    'aria-labelledby'?: string;
+    /**
+     * A string value that provides a user-friendly name for the current value of the slider.
+     */
+    'aria-valuetext'?: string;
+    /**
+     * Override or extend the styles applied to the component.
+     */
+    classes?: {};
```

Comment:
For some reason after running `yarn proptypes` + `yarn docs:api` the classes that were inside this object were removed :\ cc @eps1lon

Reply:
The until now was that every component is a styled component. This won't be an issue with the new infrastructure since we'll only have a single source of truth and not this complicated reconciliation process. 

I'll work on a fix.

Reply:
So...There's a fix but it's quite involved if we want to make sure that the old safeguards stay intact. I don't think it's wise to work on generating the documentation before we've finalized the implementation. Let's worry about this later.